pkg/frontend/querymiddleware/astmapper/rangemapper.go

```diff
@@ -0,0 +1,218 @@
+package astmapper
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+type rangeMapper struct {
```

Comment:
We need to find a better name, cause "range mapper" looks related to "range queries".

Reply:
True 🤔 What about `instantSplitterMapper`?